### PR TITLE
test: Mock server

### PR DESCRIPTION
**Pull request by lemaitre-aneo**

### CODE REVIEW

File: packages/csharp/ArmoniK.Api.Mock/ArmoniK.Api.Mock.csproj [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <Company>ANEO</Company>
+    <Copyright>Copyright (C) ANEO, 2021-2022</Copyright>
```
Comment (dbrasseur-aneo) [inline packages/csharp/ArmoniK.Api.Mock/ArmoniK.Api.Mock.csproj:6]: Note : we should update the copyright in all projects of Api

File: packages/csharp/ArmoniK.Api.Mock/CountingService.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+using MethodDecorator.Fody.Interfaces;
+
+namespace ArmoniK.Api.Mock;
+
+public static class CountingService
+{
+  /// <summary>
+  ///   Dictionary with all counts of all counting services
+  /// </summary>
+  internal static readonly Dictionary<string, Dictionary<string, StrongBox<long>>> Counts;
```
Comment (dbrasseur-aneo) [inline packages/csharp/ArmoniK.Api.Mock/CountingService.cs:35]: What's the point of the StrongBox ?

File: packages/csharp/ArmoniK.Api.Mock/CountingService.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+/// </summary>
+[AttributeUsage(AttributeTargets.Method)]
+[UsedImplicitly]
+public class CountAttribute : Attribute, IMethodDecorator
+{
+  public void Init(object     instance,
+                   MethodBase method,
+                   object[]   args)
+  {
+    _ = args;
```
Comment (dbrasseur-aneo) [inline packages/csharp/ArmoniK.Api.Mock/CountingService.cs:140]:
```suggestion
                   object[]   _)
  {
```

File: packages/csharp/ArmoniK.Api.Mock/CountingService.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    Interlocked.Increment(ref CountingService.Counts[instance.GetType()
+                                                             .Name][method.Name]
+                                             .Value);
+  }
+
+  public void OnException(Exception exception)
+    => _ = exception;
```
Comment (dbrasseur-aneo) [inline packages/csharp/ArmoniK.Api.Mock/CountingService.cs:147]:
```suggestion
  public void OnException(Exception _);
```

File: packages/csharp/ArmoniK.Api.Mock/Program.cs
Comment (dbrasseur-aneo) [inline packages/csharp/ArmoniK.Api.Mock/Program.cs:1]: Note for later : would be nice to be able to test TLS/mTLS connectivity here

File: packages/csharp/ArmoniK.Api.Mock/CountingService.cs (hunk 3 above)
Comment (lemaitre-aneo) [reply packages/csharp/ArmoniK.Api.Mock/CountingService.cs:140]: I think I prefer `_ = args;` as it does not redefine the name `_`.

**Review by dbrasseur-aneo — APPROVED** [review]

**Pull request merged**